code/game/objects/machinery/computer/nuke_disk_generator.dm

```diff
@@ -0,0 +1,170 @@
+	return PROCESS_KILL
+
+
+/obj/machinery/computer/nuke_disk_generator/attackby(obj/item/I, mob/living/user, params)
+	return attack_hand(user)
+
+/obj/machinery/computer/nuke_disk_generator/attack_hand(mob/living/user)
+	. = ..()
+	if(.)
+		return
+
+	interact(user)
+
+/obj/machinery/computer/nuke_disk_generator/interact(mob/user)
+	user.set_interaction(src)
+	var/dat = ""
+	dat += "<a href='?src=[REF(src)];mach_close=computer'>Close</a><br><br>"
+	dat += "<div align='center'><a href='?src=[REF(src)];generate=1'>Run Program</a></div>"
+	dat += "<br/>"
+	dat += "<hr/>"
+	dat += "<div align='center'><h2>Status</h2></div>"
+
+	var/message = "Error"
+	if(completed_segments >= total_segments)
+		message = "Disk generated. Run program to print."
+	else if(current_timer)
+		message = "Program running"
+	else if(completed_segments == 0)
+		message = "Idle"
+	else if(completed_segments < total_segments)
+		message = "Restart required. Please rerun the program"
+	else
+		message = "Unknown"
+
+	var/progress = round((completed_segments / total_segments) * 100)
+
+	dat += "<br/><span><b>Progress</b>: [progress]%</span>"
+	dat += "<br/><span><b>Time left</b>: [current_timer ? round(timeleft(current_timer) * 0.1, 2) : 0.0]</span>"
+	dat += "<br/><span><b>Message</b>: [message]</span>"
+
+	var/flair = ""
+	for(var/i in 1 to completed_segments)
+		flair += "[technobabble[i]]<br />"
+
+	dat += "<br /><br /><span style='font-family: monospace, monospace;'>[flair]</span>"
+
+	var/datum/browser/popup = new(user, "computer", "<div align='center'>Nuke Disk Generator</div>")
+	popup.set_content(dat)
+	popup.open()
+
+/obj/machinery/computer/nuke_disk_generator/Topic(href, href_list)
+	. = ..()
+	if(.)
+		return
+
+	if(href_list["generate"])
+		if(printing || current_timer)
+			to_chat(usr, "<span class='warning'>A program is already running.</span>")
+			return
+		if(reprintable)
+			printing = TRUE
+			addtimer(VARSET_CALLBACK(src, printing, FALSE), printing_time)
+
+			usr.visible_message("[usr] started a program to regenerate a nuclear disk code.", "You started a program to generate a nuclear disk code.")
+			if(!do_after(usr, printing_time, TRUE, src, BUSY_ICON_GENERIC, null, null, CALLBACK(src, /datum.proc/process)))
+				return
+
+			print_disc()
+			return
+
+		printing = TRUE
+		addtimer(VARSET_CALLBACK(src, printing, FALSE), segment_time)
+
+		usr.visible_message("[usr] started a program to generate a nuclear disk code.", "You started a program to generate a nuclear disk code.")
+		if(!do_after(usr, segment_time, TRUE, src, BUSY_ICON_GENERIC, null, null, CALLBACK(src, /datum.proc/process)))
+			return
+
+		current_timer = addtimer(CALLBACK(src, .proc/complete_segment), generate_time, TIMER_STOPPABLE)
+
+	updateUsrDialog()
+
+
+/obj/machinery/computer/nuke_disk_generator/proc/complete_segment()
+	playsound(src, 'sound/machines/ping.ogg', 25, 1)	
+	current_timer = null

```

Comment:
`deltimer` doesn't set the var to none which is the important bit.

Reply:
You need both then

Reply:
I dont because this function triggers on the timer so its already null, its just the timerId that is left assigned to `current_timer` that i reset for the next segment, I could add the call but its wasteful.

Reply:
If you make a TIMER_STOPPABLE you should call deltimer on it at some point, otherwise don't make it stoppable since you are never stopping it early.

Reply:
I added timer_stoppable because it just outputs a value when set, which i used to know if a timer is running, honestly i could just set TRUE, but this feels like its actually tied to something useful.

Reply:
This looks a bit like personal pref to me, not sure.
Maybe ask @ninjanomnom and see if he has a reason to enforce `TIMER_STOPPABLE` to only be used when actually stoppable, and not just because of the timer ID.

Reply:
Let's wait for ninja then